### PR TITLE
[ISSUE #2642] Allow to send ERC20 tokens

### LINKED ISSUE #2009: Wallet: do not rely on default gas
### User Story

As a developer I want to control default gas used when calling `sendTransaction`. This allows to provide better feedback to users and provide consistent experience across `web3` providers.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

#### Expected behavior

When sending ethers a warning should be displayed if user balance is lower then amount + gasPrice * gas.

### LINKED ISSUE #2642: Allow to send/request ERC20 token: wallet
### User Story

As a user, I want to send ERC20 tokens from the wallet.

### Description

Modify the wallet so that ERC20 tokens can be sent / requested.

*Type*: Feature

### LINKED ISSUE #2645: Improve QR code to add ERC20 support
### User Story

As a user, I want to be able to scan QR code containing ERC20 tokens details.

### Description

QR code can be used to encapsulate ERC20 tokens transfer. Make sure those are supported.

*Type*: Feature

### PR DESCRIPTION
**Pull request by jeluard**
fixes #2642 
fixes #2645 
fixes #2009 

### Summary:

Allow to send ERC20 tokens. Also improves QR code handling (so that token request can be encapsulated) and allow to define gas details.

### Steps to test:
- Open Status
- Navigate to wallet
- Send tokens, it should work
- Optionally modify gas details

status: ready

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

File: src/status_im/ui/screens/wallet/send/db.cljs [hunk 1]
```diff
@@ -20,9 +20,15 @@
 (spec/def ::camera-permitted? boolean?)
 (spec/def ::in-progress? boolean?)
 (spec/def ::from-chat? (spec/nilable boolean?))
+(spec/def ::symbol (spec/nilable keyword?))
+(spec/def ::gas (spec/nilable money/valid?))
+(spec/def ::gas-price (spec/nilable money/valid?))
+(spec/def ::symbol (spec/nilable keyword?))
```
Comment (dmitryn) [inline src/status_im/ui/screens/wallet/send/db.cljs]: you have same spec twice line 23 & 26

File: src/status_im/ui/screens/wallet/utils.cljs [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(ns status-im.ui.screens.wallet.utils
+  (:require [status-im.utils.money :as money]))
+
+(defn format-amount [amount decimals]
+  (money/to-fixed (money/token->unit (or amount (money/bignumber 0)) decimals)))
```
Comment (dmitryn) [inline src/status_im/ui/screens/wallet/utils.cljs]: Too many expressions nested, i find this easier to read:
```clj
(defn format-amount [amount decimals]
  (-> amount
      (or (money/bignumber 0))
      (money/token->unit decimals)
      money/to-fixed))
```

**Review by dmitryn — APPROVED** [review]

File: src/status_im/ui/screens/wallet/utils.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/ui/screens/wallet/utils.cljs]: Makes sense!

File: src/status_im/ui/screens/wallet/send/db.cljs (hunk 1 above)
Comment (jeluard) [reply src/status_im/ui/screens/wallet/send/db.cljs]: :+1:

**Comment by asemiankevich** [comment]
Branch: status-im:feature/2642
Android: https://i.diawi.com/qGJz8x
iOS: https://i.diawi.com/Ff3D2w

1. No Public key QR shown in profile
https://app.testfairy.com/projects/4803622-status/builds/7396891/sessions/2/?accessToken=i/XKYkuMtuPOE60Y09v14pzWL34

https://app.testfairy.com/projects/4803590-status/builds/7396890/sessions/1/?accessToken=EAazqP6oH4G1XUTe2md/gIbzuwY (04.40)
image

**Comment by asemiankevich** [comment]
Branch: status-im:feature/2642
Android: https://i.diawi.com/UjJSfT
iOS: https://i.diawi.com/NnsgGY

1. Request QR code throws an error:
image
2. When I /request tokens (STT) and select a contact from contact list - the request message shows ETH https://app.testfairy.com/projects/4803622-status/builds/7416911/sessions/6/?accessToken=6PrgqYOC2S7wgO2-kyGlkQaXq6g (from 2:58)
3. Can't send tokens in 1x1 chat (expected, this is known limitation)
4. Incorrect asset is shown on Advanced screen when sending tokens (ETH shown instead of STT)
image

**Comment by asemiankevich** [comment]
Branch: PR-2847
Android: https://i.diawi.com/tr2M7c
iOS: https://i.diawi.com/DpXDLf

1. Can't receive STT, pasting `0x34358c45fba99ef9b78cb501584e8cbfa6f85cef` to recipient field causes error
image
2. https://github.com/status-im/status-react/issues/2879 logged for `/request`
3. Incorrect asset is shown on Advanced screen when sending tokens (ETH shown instead of STT) - it is fixed
image

**Comment by asemiankevich** [comment]
Branch: PR-2847
Android: https://i.diawi.com/jKyaM2
iOS: https://i.diawi.com/7NGL4w

1. Can claim for tokens both android and ios
2. Error shown when I scan QR code generated by tapping Request button in Wallet. 
STR: 

- 1. Generate Request QR code on one device
- 2. Scan the code with another device

 image

image

3. I can send tokens from wallet (select contact from list, scan QR code from profile)
4. Can receive tokens when they are sent (asset balance is updated)
5. STT transactions are listed as ETH in history and etherscan, so no indication that I have sent tokens - will be in separate PR
6. https://github.com/status-im/status-react/issues/2879 opened separately
7. Observing some UI glitches that will be fixed by another swarm (#61)

**Comment by asemiankevich** [comment]
Branch: PR-2847
Android: https://i.diawi.com/6tpgoV
iOS: https://i.diawi.com/CZYCZK

1. Can claim for tokens both android and ios
2. No QR code errors anymore
3. I can send tokens from wallet (select contact from list, scan QR code from profile, scan QR from wallet-request)
4. Can receive tokens when they are sent (asset balance is updated)
5. History will be implemented in separate PR
6. https://github.com/status-im/status-react/issues/2879 opened separately
7. Observing some UI glitches that will be fixed by another swarm (#61)

This PR is good to merge

**Pull request merged**